app/uk/gov/hmrc/pensionschemereturnsipp/services/SippPsrSubmissionService.scala

```diff
@@ -549,4 +550,112 @@ class SippPsrSubmissionService @Inject()(
         case None =>
           Future.failed(new Exception(s"Submission with pstr $pstr not found"))
       }
+
+  def deleteAssets(
+    journey: Journey,
+    journeyType: JourneyType,
+    pstr: String,
+    optFbNumber: Option[String],
+    optPeriodStartDate: Option[String],
+    optPsrVersion: Option[String],
+    pensionSchemeId: PensionSchemeId
+  )(
+    implicit hc: HeaderCarrier,
+    requestHeader: RequestHeader
+  ): Future[Unit] =
+    psrConnector
+      .getSippPsr(pstr, optFbNumber, optPeriodStartDate, optPsrVersion)
+      .flatMap {
+        case Some(response) =>
+          val updatedMembers = response.memberAndTransactions.flatMap { mTxs =>
+            NonEmptyList.fromList(mTxs.map(mTx => deleteAssetForJourney(journey, mTx)))
+          }
+
+          val updateRequest = SippPsrSubmissionEtmpRequest(
+            reportDetails = response.reportDetails.copy(status = EtmpPsrStatus.Compiled, version = None),
+            accountingPeriodDetails = response.accountingPeriodDetails,
+            memberAndTransactions = updatedMembers,
+            psrDeclaration = response.psrDeclaration.map(
+              declaration =>
+                declaration.copy(
+                  psaDeclaration =
+                    declaration.psaDeclaration.map(current => current.copy(declaration1 = false, declaration2 = false)),
+                  pspDeclaration =
+                    declaration.pspDeclaration.map(current => current.copy(declaration1 = false, declaration2 = false))
+                )
+            )
+          )
+
+          submitWithRequest(journeyType, pstr, pensionSchemeId, Future.successful(updateRequest)).map(_ => ())
+
+        case None =>
+          Future.failed(new Exception(s"Submission with pstr $pstr not found"))
+      }
+
+  private def deleteAssetForJourney(journey: Journey, member: EtmpMemberAndTransactions): EtmpMemberAndTransactions = {
+    var assetDeleted = false
+    if (member.status == Deleted)
+      member
+    else {
+      // First, delete the specific asset based on the journey
+      val updatedMember = journey match {
+        case Journey.InterestInLandOrProperty =>
+          if (member.landConnectedParty.isDefined) {
```

Comment:
optional: perhaps you could `member.landConnectedParty.map( m => m.update...)` and return an optional member rather than use `assetDeleted` var ?